### ISSUE TITLE
Only Purge tables that have a fixture class

### ISSUE THREAD

**Issue by BladeRavinger** [doctrine/data-fixtures#311]
So im obviously working on my website and for back end trial and error junk i require user logins to get at admin sections for reasons. it would be nice if instead of a database purge, wiping my 3 users of various permission levels, it would only drop/create tables that fixtures will be working on, as at present i am not creating dummy user data and its a pain to recreate the users each time i reload fixtures.

**Comment by thewind1984** [comment]
> Purging only tables based on the fixture classes is going to be difficult: there’s nothing stopping you from creating multiple different entities in a single fixture class, so the library can’t really know what to clear based on the class.
> 
> I’m moving this issue to the data-fixtures project, as this is nothing that should be solved in the bundle, but rather in the underlying library.

Maybe the following idea could be fixable - to add new option `-exclude="table1, table2, ..."` to prevent some tables to be truncated...

**Comment by BladeRavinger** [comment]
Perhaps the fact that in the fixture class we `use App\Entity\Foo;`
don't affect tables that are not used?

obviously if you try to make changes without a use statement you get errors
so as its a thing that will exist, and must exist, can we not look for `use App/Entity/*` and drop or truncate these automatically. i havent a clue if this is even possible, but as a default behavior that in itself makes sense. 

thats just my 10cent. 

i thank you all for the help.

**Comment by alcaeus** [comment]
Note that currently the purger has no idea of what you’re doing in your fixtures. I’m not saying it’s impossible, but it’s also not something that we can just fix. It definitely requires some thinking as to how to accomplish it.

**Comment by BladeRavinger** [comment]
> Note that currently the purger has no idea of what you’re doing in your fixtures. I’m not saying it’s impossible, but it’s also not something that we can just fix. It definitely requires some thinking as to how to accomplish it.

of course, being a sub par coder myself i understand this, so the idea is here if you guys ever choose to work on it. im not pushing you to go make the change. Thewind1984 gave me a solution that will work for what i want, but the suggestion still stands if you guys choose to incorporate it in the future

**Comment by vincentjdc** [comment]
This is not the first time I would like to have a feature like the one describe here.

A common use, is when you have a table for statuses of another table (like "Project" and "ProjectStatus").
In my case, ProjectStatus can't be added using the app. So to manage those statuses, I use Fixtures.

When pushing in prod, I need to update the prod ProjectStatus in database values but I just can't load the ProjectStatusFixtures beacause the only two options are :
- append data which results in duplicated values
- purge all the data with data loss in other tables

A solution could be an interface `Doctrine\DataFixtures\TableRelatedFixtureInterface` with a method `public static function getTables(): array`.
When implementing the interface, the developer needs to define that method to give the information of which tables are affected by the fixture (and should be purged)
And option like `--purge-only-related-tables` will make the purger to check the `getTables()` of each Fixtures file.
